### PR TITLE
Added Banner events

### PR DESCRIPTION
**Pull request by avikalpg**
Track event will be triggered in two cases:
1. On every page load that detects that our extension is not installed (or browser/device is incompatible with browser extensions)
2. When the "download" button is clicked in the banner to install the browser extension.

Other changes:
- Changed the copy for incompatible devices (asking them to use the website on a computer).
- Navbar bug fixes: the mobile menu didn't have the `id` attribute for eventing. And the "Pricing" link was dead for mobile.

### CODE REVIEW

File: components/LoginLogout.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { useEffect, useState, useContext } from "react";
 import type { Session } from 'next-auth';
 import Link from 'next/link';
 import RudderContext from './RudderContext';
-import { getAndSetAnonymousIdFromLocalStorage } from '../utils/url_utils';
+import { getAndSetAnonymousIdFromLocalStorage } from '../utils/rudderstack_initialize';
```
Comment (avikalpg) [inline components/LoginLogout.tsx:8]: I am not sure why GitHub is showing these changes here. I have double checked that both the branches, `akg/demo_banner` and `akg/banner_events` have these changes from master. In fact, I have branched out `akg/banner_events` after I had pulled main into `akg/demo_banner`. 

Please ignore these.

File: components/Banner.tsx [hunk 2]
```diff
@@ -66,20 +74,39 @@ const Banner = ({ bannerHeight, setBannerHeight }: {
 		}
 
 		const determineSituation = (): BannerSituation => {
-            const isUnsupportedDevice = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
+			const isUnsupportedDevice = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
```
Comment (avikalpg) [inline components/Banner.tsx:77]: @doomsday4 These whitespace changes are the conversion of spaces into tabs.
I suggest that you change your VSCode settings to automatically detect the indentation from the file. This file has tab-indentation throughout, except for the lines that you added.

File: components/Banner.tsx [hunk 3]
```diff
@@ -74,10 +82,29 @@ const Banner = () => {
 				return "incompatible-browser";
 			}
 		}
+
 		const situation = determineSituation();
+		if (situation) {
+			const anonymousId = getAndSetAnonymousIdFromLocalStorage();
+			rudderEventMethods?.track(getAuthUserId(session), situation, { type: "detection", eventStatusFlag: 0, source: "banner", name: getAuthUserName(session) }, anonymousId);
+		}
```
Comment (MuskanPaliwal) [inline components/Banner.tsx:90]: This event is being sent with `eventstatusflag` as `0`?

**Review by MuskanPaliwal — APPROVED** [review]

File: components/Banner.tsx [hunk 4]
```diff
@@ -2,19 +2,27 @@ import Image from "next/image";
...
 import chromeLogo from '../public/chrome-logo.png'
+import { useSession } from "next-auth/react";
+import type { Session } from "next-auth";
+import RudderContext from "./RudderContext";
+import { getAndSetAnonymousIdFromLocalStorage } from "../utils/rudderstack_initialize";
+import { getAuthUserId, getAuthUserName } from "../utils/auth";
 
 export type BannerHeightType = "h-12" | "h-24" | "h-32" | "h-40" | "h-44" | undefined;
-type BannerSituation = "not-installed" | "incompatible-browser" | "incompatible-device" | null;
+type BannerSituation = "extension-not-installed" | "incompatible-browser" | "incompatible-device" | null;
```
Comment (MuskanPaliwal) [inline components/Banner.tsx:12]: Please don't change these types since the event name has been decided based on these types only.

**Review by MuskanPaliwal — APPROVED** [review]

**Review by MuskanPaliwal — APPROVED** [review]

**Pull request merged**